src/MudBlazor/Components/Dialog/MudDialogProvider.razor.cs

```
@@ -152,8 +153,8 @@ internal void DismissInstance(Guid id, DialogResult result)
 
         private Task AddInstanceAsync(IDialogReference dialog)
         {
-            _dialogs.Add(dialog);
-
+            if (!_dialogs.TryAdd(dialog.Id, dialog))
+                throw new InvalidOperationException("Guid dialog already exists.");
```

@henon do we want to thow exception?

---

Since it is the `DialogProvider` that creates the dialog and immediately adds it to the dictionary, this case should not occur.

But I added this exception to help the librairy developper to detect a problem. So [`Debug.Assert`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert) would be better?

---

> this case should not occur.

Yeah, such a situation shouldn't occur since the ID is generated on our side when `CreateReference` is called. But if someone wrote something custom on top of this API, then I don't know. That's why I'm not sure whether we should throw an error, ignore, or maybe even update the IDialogReference with this ID. That's why I need Henon's opinion.

---

I think it is ok to throw for now. If later someone requires different behavior and can argue their point we can change it.

@vernou please change the message to `"A dialog with this id already exists"`

---

Done